Document Tweaks for running Teleport at scale. (#4659)
gravitational/teleport: docs/4.4/admin-guide.md

We'll cover how to use `etcd`, DynamoDB and Firestore storage back-ends to make Teleport
highly available below.

### Teleport Scalability Tweaks

> When running Teleport at scale or in case of 10K+ IOT clusters, the following
> settings should be set on Teleport auth and proxies:

Reviewer: ```suggestion
When running Teleport at scale (for example in the case where there are 10,000+ nodes connected
to a cluster via [node tunnelling mode](#adding-a-node-located-behind-nat), the following settings
should be set on Teleport auth and proxies:
```

> #### Proxy Server

Reviewer: ```suggestion
#### Proxy Servers
```

> These setting update Teleports [Default Connections from 15000](https://github.com/gravitational/teleport/blob/5cd212fecda63ec6790cc5ffe508a626c56e2b2c/lib/defaults/defaults.go#L385) to 65000.

Reviewer: ```suggestion
These settings alter Teleport's [default connection limit](https://github.com/gravitational/teleport/blob/5cd212fecda63ec6790cc5ffe508a626c56e2b2c/lib/defaults/defaults.go#L385) from 15000 to 65000.
```

```yaml
# Teleport Proxy
teleport:
  cache:

>     # use in memory cache to speed up connections of many teleport nodes

Reviewer: ```suggestion
    # use an in-memory cache to speed up the connection of many teleport nodes
```

    # back to proxy
    type: in-memory

>   # set up connection limits to prevent throttling of many iot nodes connecting to proxies

Reviewer: ```suggestion
  # set up connection limits to prevent throttling of many IoT nodes connecting to proxies
```

  connection_limits:
    max_connections: 65000
    max_users: 1000
```

> #### Auth Server

Reviewer: ```suggestion
#### Auth Servers
```
